### ISSUE TITLE
`cast::Error` does not implement `Error` trait

### ISSUE THREAD

**Issue by emk** [japaric/cast.rs#8]
Hello! I'm using the `cast` library in some production code, and it's great! I like the way that casts only return `Result` if they might fail.

I'm also using the [error-chain](https://github.com/brson/error-chain) crate, which provides support for converting many different error types into my library's "official" error type. But for this to work, all error types need to implement the `Display` and `Error` traits. Would you be interested in a pull request adding support for this to your `cast` library?

Thank you for considering this!

**Comment by emk** [comment]
As a workaround, people using `error-chain` can add something like the following their `errors` module:

```
impl From<cast::Error> for Error {
    fn from(err: cast::Error) -> Error {
        format!("numeric cast error: {:?}", err).into()
    }
}
```

**Comment by japaric** [comment]
> Would you be interested in a pull request adding support for this to your cast library?

Sure! :shipit:

**Comment by emk** [comment]
Could you please assign this to me using the Issues interface so that I don't lose track of it? Thank you! I'm juggling a bunch of Rust PRs the next couple of days.

**Comment by japaric** [comment]
@emk Done!

**Comment by emk** [comment]
OK, I've submitted the PR #9, which should provide `Error` (and several other useful `impl`s). There's an interesting complication because `Error` requires `std` and not `core`, but I provided a way to build the library either with or without `core`. See the PR for details.

Thank you for your quick response the other day! This is a very useful crate, and I'm going to encourage my coworkers at Faraday to make this part of our standard coding style!
